docs/guides/src/main/server/containers.adoc

```diff
@@ -62,6 +62,45 @@ To install custom providers, you just need to define a step to include the JAR f
 RUN curl -sL <MY_PROVIDER_JAR_URL> -o /opt/keycloak/providers/myprovider.jar
 ----
 
+=== Installing additional RPM packages
+
+If you try to install new software in a stage `+FROM quay.io/keycloak/keycloak+`, you will notice that `+microdnf+`, `+dnf+`, and even `+rpm+` are not installed. Also, very few packages are available, only enough for a `+bash+` shell, and to run Keycloak itself. This is due to security hardening measures, which reduce the attack surface of the Keycloak container.
+
+First, consider if your use case can be implemented in a different way, and so avoid installing new RPMs into the final container:
+
+* A `+RUN curl+` instruction in your Dockerfile can be replaced with `+ADD+`, since that instruction natively supports remote URLs.
+* Some common CLI tools can be replaced by creative use of the Linux filesystem. For example, `+ip addr show tap0+` becomes `+cat /sys/class/net/tap0/address+`
+* More advanced tools can be replaced with a custom executable jar. Java is already available in the image for Keycloak's use.
```

Comment:
TBH, I'm not sure we should instruct users to re-write they favorite tools in Java. ;) Even from the security perspective, I think the lesser evil is to use some maintained RPM package rather than writing it from scratch.

I'd remove this bullet point.

Reply:
It's apparent from the other thread that there are users doing all sorts of weird and wonderful things with their custom containers. Fortunately for us, that stuff is all their responsibility, provided we don't suggest they install more RPMs. I suppose just omitting an alternative is ok too.

Reply:
Yeah, so then agreed on removing the jar alternative then?